### ISSUE TITLE
Dead accent/single quote key problem with Spanish keyboard

### ISSUE THREAD

**Issue by mcamou** [sorin-ionescu/prezto#859]
The Spanish keyboard layout doesn't have apostrophe (single quote) or backquote keys  - it uses the accent dead keys, if you press a key that is not an accented letter after them it gets turned into the proper quote type followed by the letter you typed. The backquote is working fine, but since I installed Prezto, the single quote is not working at all (not even for generating accented characters).

**Comment by sorin-ionescu** [comment]
Have you changed your terminal meta key settings?

**Comment by mcamou** [comment]
I haven't changed them.

Under bash, or under zsh without Prezto, if I hit the dead key I get a
single quote directly (i.e. it doesn't function as a dead key as in other
applications). The other dead keys in the keyboard work correctly in all
cases (bash, zsh without Prezto and zsh with Prezto).

I really don't mind the dead key not working as a dead key (since I very
rarely type accented characters in the shell), but not having the single
quote is a big problem.


-Mario.

--
I want to change the world but they won't give me the source code.

**Comment by sorin-ionescu** [comment]
Look through modules/editor/init.zsh to find what's causing it.

**Comment by mcamou** [comment]
OK, so now I have a Mac with a US keyboard, and I'm experiencing the same problem. The single-quote key doesn't work at all. I had assumed it might be a problem with the Spanish keyboard, but no.

I've tried disabling all modules, I've looked through modules/editor/init.zsh, and I can't find a possible cause for this. I am using the built-in Mac terminal (not iTerm). The key works correctly without Prezto enabled.

Any ideas where else to look?

**Comment by sorin-ionescu** [comment]
I do not know. You'll have to comment-out lines. Does it work in iTerm2?

**Comment by mcamou** [comment]
I just downloaded and installed iTerm2, and the problem is there too. The thing is, if I disable prezto, it works fine. I am surprised that no one has bumped into this before.

**Comment by PaBLoX-CL** [comment]
I'm using prezto on a latam layout, and works fine. I changed to spanish layout to check if the problem happen there, but it doesn't. The single quote (<kbd>'</kbd>works fine, and it's not a dead key... it's the key to the right of the zero. The backquote it is a dead key (right to the <kbd>p</kbd>), but works fine double tapping it, or pressing the dead key followed by space.

http://mylingos.com/keyboards/spanishspain.html

**Comment by mcamou** [comment]
I've finally found the issue. On first run, zprezto asked me to press different keys to get my keymap. The problem was that my keyboard did not generate Ctrl+Left or Ctrl+Right, so the following lines were added to the corresponding .zkbd file:

```
key[Control-Left]=''''
key[Control-Right]=''''
```

Note the quadruple single-quotes. Editing the file and changing the quadruple single-quotes to double single-quotes fixed the problem. Something in the first run configurator messed things up.

Just leaving this here in case someone else bumps up against this (or if you want to have a look and see if it's a bug).

Thanks!
